CoordinateSystemsCore/src/transition_map.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    transition_map(to::CoordinateSystem, from::CoordinateSystem)
+
+Get the transition map from chart `from` to chart `to`.
+
+# Examples
+
+```jldoctest
+julia> mapfn = transition_map(Cartesian{2}(), Spherical{2}());
+
+julia> mapfn([1, pi/2]) ≈ [0, 1]
+true
+
+julia> mapfn = transition_map(Spherical{2}(), Cartesian{2}());
+
+julia> mapfn([0, 1]) ≈ [1, pi/2]
+true
+```
+"""
+transition_map
+
+@inline transition_map(to::CT, from::CT) where CT<:CoordinateSystem = identity
+
+@inline function transition_map(to::Cartesian{N}, from::Spherical{N}) where N
+    Base.@propagate_inbounds function Spherical_to_Cartesian(x::AbstractVector{T}) where T
```

Comment:
A minor thing - I think it would be clearer here if `Spherical_to_Cartesian` were just pulled out into a normal function (rather than being a closure). Then return that from `transition_map`.

Reply:
Agreed, I've thought about this too and maybe even make CoordinateSystem a functor.
This would add more additional mess to this PR so I might work it in the next PR.